### PR TITLE
VPN NAT for Virtual WAN feature changes

### PR DESCRIPTION
**Pull request by Nilambari**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - { https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/790 }
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by Nilambari** [comment]
Related merged swagger changes :- Azure/azure-rest-api-specs#11815
Related approved cmdlet design review :- https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/790

**Comment by nilambarideshpande** [comment]
@wyunchi-ms Can you take a look at the PR? Thanks!

**Comment by wyunchi-ms** [comment]
1. Please fix the issue of online version in new generated help files.
2. Please fix the test issues of other modules whoes test project dependence on Network SDK.

**Comment by Nilambari** [comment]
> 1. Please fix the issue of online version in new generated help files.
> 2. Please fix the test issues of other modules whoes test project dependence on Network SDK.

Thanks Yunchi for taking a look. Incorporating comments.

**Comment by Nilambari** [comment]
@MikhailTryakhov  Please take a look as I need to add unsigned sdk in all test projects, otherwise build is failing.

**Comment by Nilambari** [comment]
@MikhailTryakhov  @wyunchi-ms  Updated with importing latest network nugit package. Please help take a look and approve the PR.

File: src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs [hunk 1]
```diff
@@ -73,6 +74,16 @@ public class NewAzVpnSiteLinkConnectionCommand : NetworkBaseCmdlet
             HelpMessage = "Use policy based traffic selectors for this link connection.")]
         public SwitchParameter UsePolicyBasedTrafficSelectors { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of ingress NAT rules that are associated with this link Connection.")]
+        public PSResourceId[] IngressNatRules { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs:80]: please re-name to IngressNatRule

File: src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs [hunk 2]
```diff
@@ -73,6 +74,16 @@ public class NewAzVpnSiteLinkConnectionCommand : NetworkBaseCmdlet
...
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The list of egress  NAT rules that are associated with this link Connection.")]
+        public PSResourceId[] EgressNatRules { get; set; }
```
Comment (VeryEarly) [inline src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs:85]: please re-name to EgressNatRule

**Review by VeryEarly — COMMENTED** [review]
please update these two parameter names,
update testcases and regenerate help

**Comment by MikhailTryakhov** [comment]
@Nilambari we won't be able to merge it today due to PR is not ready yet

File: src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs (hunk 1 above)
Comment (Nilambari) [reply src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs:80]: Done.

File: src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs (hunk 2 above)
Comment (Nilambari) [reply src/Network/Network/Cortex/VpnConnection/NewAzVpnSiteLinkConnectionCommand.cs:85]: Done.

**Comment by Nilambari** [comment]
Closing this PR as its against Azure:network-august branch which is merged to master yesterday.
Opened new PR:-https://github.com/Azure/azure-powershell/pull/13943 against Azure:master branch.

**Comment by Nilambari** [comment]
Closing this PR as its against Azure:network-august branch which is merged to master yesterday.
Opened new PR:-#13943 against Azure:master branch.

**Pull request closed without merging**